ignite/metrics/ssim.py

```diff
@@ -102,12 +103,13 @@ def __init__(
         self.c2 = (k2 * data_range) ** 2
         self.pad_h = (self.kernel_size[0] - 1) // 2
         self.pad_w = (self.kernel_size[1] - 1) // 2
-        self._kernel = self._gaussian_or_uniform_kernel(kernel_size=self.kernel_size, sigma=self.sigma)
+        self._kernel_2d = self._gaussian_or_uniform_kernel(kernel_size=self.kernel_size, sigma=self.sigma)
 
     @reinit__is_reduced
     def reset(self) -> None:
         self._sum_of_ssim = torch.tensor(0.0, dtype=torch.float64, device=self._device)
         self._num_examples = 0
+        self._kernel = torch.empty(0)
```

Comment:
I liked the previous code with `self._kernel = None` in the ctor.

Reply:
Me too, but mypy didn't and was throwing errors :(